src/pal/inc/pal.h

```diff
@@ -5683,7 +5685,8 @@ struct PAL_SEHException
 
     void FreeRecords()
     {
-        if (ExceptionPointers.ExceptionRecord != NULL)
+        if (ExceptionPointers.ExceptionRecord != NULL &&
+            ! (ExceptionPointers.ExceptionRecord->ExceptionFlags | EXCEPTION_ON_STACK) )
```

Comment:
This second part of the condition is not correct, it should be `&` instead of `|`.